src/engraving/dom/cmd.cpp

```diff
@@ -2075,6 +2075,114 @@ void Score::toggleArticulation(SymId attr)
     }
 }
 
+//---------------------------------------------------------
+//   toggleDynamic
+//---------------------------------------------------------
+
+void Score::toggleDynamic(DynamicType dt)
+{
+    std::set<Chord*> set;
+
+    for (EngravingItem* el : selection().elements()) {
+        if (el->isNote() || el->isChord()) {
+            Chord* cr = 0;
+            // apply dynamic to first chord/note in selection
+            if (el->isNote()) {
+                cr = toNote(el)->chord();
+                if (mu::contains(set, cr)) {
+                    continue;
+                }
+            }
+            /*Dynamic* dy = Factory::createDynamic(cr->segment());
+            dy->setDynamicType(dt);
+            dy->setTrack(cr->track());
+            cr->segment()->add(dy);*/
+
+
+            //undoAddElement(dy);
+ 
+            Articulation* na = Factory::createArticulation(this->dummy()->chord());
```

Comment:
It seems that you are abusing Articulations to show dynamics. Why not use the `Dynamic` class?